widgetssdk/src/main/java/com/glia/widgets/filepreview/data/source/local/DownloadsFolderDataSource.java

```diff
@@ -0,0 +1,95 @@
+package com.glia.widgets.filepreview.data.source.local;
+
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Environment;
+import android.provider.MediaStore;
+
+import com.glia.widgets.helper.Logger;
+import com.squareup.picasso.Picasso;
+import com.squareup.picasso.Target;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Objects;
+
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
+
+public class DownloadsFolderDataSource {
+    private final static String TAG = DownloadsFolderDataSource.class.getSimpleName();
+    private final Context context;
+
+    public DownloadsFolderDataSource(Context appContext) {
+        this.context = appContext;
+    }
+
+    public Maybe<Bitmap> getImageFromDownloads(String imageName) {
+        return Maybe.create(
+                emitter -> {
+                    File imageFile = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).toString(), imageName);
+                    Picasso.get()
+                            .load(imageFile)
+                            .into(new Target() {
+                                @Override
+                                public void onBitmapLoaded(Bitmap bitmap, Picasso.LoadedFrom from) {
+                                    emitter.onSuccess(bitmap);
+                                }
+
+                                @Override
+                                public void onBitmapFailed(Exception e, Drawable errorDrawable) {
+                                    emitter.onError(e);
+                                }
+
+                                @Override
+                                public void onPrepareLoad(Drawable placeHolderDrawable) {
+
+                                }
+                            });
+                }
+        );
+    }
+
+    public Completable putImageToDownloads(String fileName, Bitmap bitmap) {
```

Comment:
function body can be divided in to different functions

Reply:
will not touch it atm - will separate to API related methods later